### PR TITLE
Post-submit test for Hosted/MKP (mpdev verify)

### PR DESCRIPTION
**Pull request by rmgogogo**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/3193)

### CODE REVIEW

File: test/cloudbuild/mkp_verify.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+steps:
+- id:   'pullDeployer'
+  name: 'gcr.io/cloud-builders/docker'
+  args: ['pull', 'gcr.io/$PROJECT_ID/hosted/$COMMIT_SHA:${_MM_VERSION}']  
+- id: "verify"
+  waitFor: ['pullDeployer']
```
Comment (Bobgy) [inline test/cloudbuild/mkp_verify.yaml]: When this is a simple sequential job, there's no need to write `waitFor`

File: test/cloudbuild/mkp_verify.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      gcloud container clusters get-credentials hosted-verify-test --zone us-central1-c --project ml-pipeline-test
+      kubectl apply -f "https://raw.githubusercontent.com/GoogleCloudPlatform/marketplace-k8s-app-tools/master/crd/app-crd.yaml"
+      /scripts/verify --deployer="gcr.io/$PROJECT_ID/hosted/$COMMIT_SHA:${_MM_VERSION}" --wait_timeout="1800"
+     
+timeout: '3600s'
+options:
+  machineType: N1_HIGHCPU_8
+
+substitutions:
+  _MM_VERSION: ''
```
Comment (Bobgy) [inline test/cloudbuild/mkp_verify.yaml]: What does MM version mean?

File: test/deploy-pipeline-mkp-cli.sh [hunk 3]
```diff
@@ -14,91 +14,10 @@
...
-echo "Status of pods after mpdev install"
-kubectl get pods -n ${NAMESPACE}
-
-popd
-
-# Waiting for the KFP resources are ready. TODO: verification of KFP resources
-sleep 60
+# sync trigger to avoid wait
+gcloud builds submit --config=test/cloudbuild/mkp_verify.yaml --substitutions=_MM_VERSION="$VERSION",COMMIT_SHA="$COMMIT_SHA" --project=ml-pipeline-test
+return $?
```
Comment (Bobgy) [inline test/deploy-pipeline-mkp-cli.sh]: I wonder why we don't set it as `set -ex` and remove this one?

File: test/tag_for_hosted.sh [hunk 4]
```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
```
Comment (Bobgy) [inline test/tag_for_hosted.sh]: nit: why did we move this file?
It seems previous location also looks good.

**Comment by rui5i** [comment]
/lgtm

Thanks, Renmin!

File: test/cloudbuild/mkp_verify.yaml (hunk 1 above)
Comment (rmgogogo) [reply test/cloudbuild/mkp_verify.yaml]: I remember hit the issue saying no such image and it didn't pull image.
you may correct me. 

prefer keep it.

By the way, do you know how to test such post-submit before a submit? via calling script directlly? (setup ADC etc. first)

File: test/cloudbuild/mkp_verify.yaml (hunk 2 above)
Comment (rmgogogo) [reply test/cloudbuild/mkp_verify.yaml]: major minor version

File: test/deploy-pipeline-mkp-cli.sh (hunk 3 above)
Comment (rmgogogo) [reply test/deploy-pipeline-mkp-cli.sh]: done

File: test/tag_for_hosted.sh (hunk 4 above)
Comment (rmgogogo) [reply test/tag_for_hosted.sh]: feel here is better as it's for testing now.
After Yang is OK we use uniform name, we don't need this file, all will be controlled in cloud build yaml.

File: test/cloudbuild/mkp_verify.yaml (hunk 2 above)
Comment (Bobgy) [reply test/cloudbuild/mkp_verify.yaml]: Thanks!

Comment (rmgogogo) [reply test/cloudbuild/mkp_verify.yaml]: changed to a more general name.

File: test/cloudbuild/mkp_verify.yaml (hunk 1 above)
Comment (rmgogogo) [reply test/cloudbuild/mkp_verify.yaml]: I put the notes to comments on how to test the changes before submit

**Comment by rmgogogo** [comment]
ready to be submitted. I tested it via manually launch them locally.

**Comment by Bobgy** [comment]
Thanks!
/lgtm
/approve

**Comment by rmgogogo** [comment]
updated VERSION to 0.3.0 as Ajay did it in another PR.
self LGTM as pass all test and locally maunal tested the new yaml

**Comment by Bobgy** [comment]
/hold
@rmgogogo Sorry, I think this is a little risky to be merged during release, because it has changes in .cloudbuild.yaml. https://github.com/kubeflow/pipelines/pull/3193/files#diff-ea967cf037de3503a92911b5c390b596

Can we wait for current release to finish first?

File: test/tag_for_hosted.sh [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+# Later we need to unify the image names and path, after that, we may don't need this file.
+# All will be handled in Cloud Build yaml file.
+PROJECT_ID=$1
+COMMIT_SHA=$2
+SEM_VER=$3
+MM_VER=$4
+
+docker tag gcr.io/$PROJECT_ID/frontend:$COMMIT_SHA gcr.io/$PROJECT_ID/hosted/$COMMIT_SHA/frontend:$SEM_VER
+docker tag gcr.io/$PROJECT_ID/frontend:$COMMIT_SHA gcr.io/$PROJECT_ID/hosted/$COMMIT_SHA/frontend:$MM_VER
+docker push gcr.io/$PROJECT_ID/hosted/$COMMIT_SHA/frontend:$SEM_VER
```
Comment (Ark-kun) [inline test/tag_for_hosted.sh:29]: Can we use `gcloud container images add-tag` here? It might be faster.

**Pull request merged**